lib/url.js

```diff
@@ -360,10 +360,18 @@ Url.prototype.parse = function parse(url, parseQueryString, slashesDenoteHost) {
       start = atSign + 1;
     }
     if (nonHost === -1) {
+      if (invalid !== -1) {
+        throw new ERR_INVALID_URL(url);
+      }
       this.host = rest.slice(start);
       rest = '';
     } else {
+      if (invalid > -1 && invalid < nonHost) {
+        throw new ERR_INVALID_URL(url);
+      }
       this.host = rest.slice(start, nonHost);
+      // WHATWG URL removes tabs, newlines, and carriage returns. Let's do that too.
+      this.host = this.host.replaceAll(/[\t\n\r]/g, '');
```

Comment:
Nit: Afaik, `string.replaceAll` is slower than `string.replace`. (Referencing: https://github.com/RafaelGSS/nodejs-bench-operations/blob/main/RESULTS-v18.md#replace-vs-replaceall-comparison)

Reply:
```suggestion
      this.host = this.host.replace(/[\t\n\r]/g, '');
```

Reply:
What's likely even more performant: use a `host` variable that contains an empty string above and concat all chars in the above switch case besides `\r`, `\n` and `\t` (the latter is already checked for). That way there's no need to check all characters twice.